common/arch/sdl/window.cpp

```diff
@@ -25,18 +25,18 @@ namespace dcx {
 static window *FrontWindow = NULL;
 static window *FirstWindow = NULL;
 
-window *window_create(grs_canvas *src, int x, int y, int w, int h, window_subfunction<void> event_callback, void *data, const void *createdata)
+window::window(grs_canvas *src, int x, int y, int w, int h, window_subfunction<void> event_callback, void *data, const void *createdata)
```

Comment:
This could, and probably should, use a *member initializer list*.  I can convert it to that after you push, to save another round of review.